src/v2/spec.yaml

```diff
@@ -988,6 +988,126 @@ components:
       properties:
         available:
           type: boolean
+    NeynarFrame:
+      type: object
+      properties:
+        uuid:
+          type: string
+          format: uuid
+          description: Unique identifier for the frame.
+        name:
+          type: string
+          description: Name of the frame.
+        link:
+          type: string
+          format: uri
+          description: Generated link for the frame's first page.
+        pages:
+          type: array
+          items:
+            $ref: '#/components/schemas/NeynarFramePage'
+        valid:
+          type: boolean
+          description: Indicates if the frame is valid.
+      required:
+        - uuid
+        - name
+        - pages
+        - link
+    NeynarFramePage:
+      type: object
+      properties:
+        uuid:
+          type: string
+          format: uuid
+          description: Unique identifier for the page.
+        version:
+          type: string
+          description: The version of the page schema.
+        title:
+          type: string
+          description: The title of the page.
+        image:
+          $ref: '#/components/schemas/NeynarFrameImage'
+        buttons:
+          type: array
+          items:
+            $ref: '#/components/schemas/NeynarFrameButton'
+        input:
+          $ref: '#/components/schemas/NeynarFrameInput'
+      required:
+          - uuid
+          - version
+          - title
+          - image
+          - buttons
+          - input
+    NeynarFrameImage:
+      type: object
+      properties:
+        url:
+          type: string
+          format: uri
+          description: The URL of the page's image.
+        aspect_ratio:
+          type: string
+          description: The aspect ratio of the image.
+      required:
+        - url
+        - aspect_ratio
+    NeynarFrameButton:
+      type: object
+      properties:
+        title:
+          type: string
+          description: The title of the button.
+        action_type:
+          type: string
+          description: The type of action that the button performs.
+        index:
+          type: integer
+          description: The index of the button.
+        next_page:
+          oneOf:
+            - $ref: '#/components/schemas/NeynarFrameNextPage'
+            - $ref: '#/components/schemas/NeynarFrameRedirect'
+      required:
+        - title
+        - action_type
+        - index
+        - next_page
+    NeynarFrameInput:
```

Comment:
wondering if page properties should be called NeynarPageInput and NeynarPageButton and so on... given that they're elements of a page vs. a NeynarFrame. wdyt?